data_structures/src/mainnet_validations.rs

```diff
@@ -16,6 +18,169 @@ pub const FIRST_HARD_FORK: Epoch = 192000;
+        // In order to not update bit counter from old blocks where the block version was not used,
+        // the first time (bit_tapi_counter.last_epoch == 0) would be skipped in this conditional branch
+        if self.bit_tapi_counter.last_epoch != 0
+            && epoch_to_update > self.bit_tapi_counter.last_epoch + 1
+        {
+            let init = self.bit_tapi_counter.last_epoch + 1;
+            let end = epoch_to_update;
+            for i in init..end {
+                self.update_bit_counter(0, i, block_epoch, avoid_wip_list);
+            }
+        }
+        for n in 0..self.bit_tapi_counter.len() {
+            if let Some(mut bit_counter) = self.bit_tapi_counter.get_mut(n, &epoch_to_update) {
+                if !self.wip_activation.contains_key(&bit_counter.wip)
+                    && !avoid_wip_list.contains(&bit_counter.wip)
+                {
+                    if is_bit_n_activated(v, n) {
+                        bit_counter.votes += 1;
+                    }
+                    if (epoch_to_update - bit_counter.init) % bit_counter.period == 0 {
+                        if (bit_counter.votes * 100) / bit_counter.period >= 80 {
+                            // An offset of 21 is added to ensure that the activation of the WIP is
+                            // achieved with consolidated blocks
+                            self.wip_activation
+                                .insert(bit_counter.wip.clone(), block_epoch + 21);
+                        }
+                        bit_counter.votes = 0;
+                    }
+                }
+            }
+        }
+        self.bit_tapi_counter.last_epoch = epoch_to_update;
+    }
+
+    pub fn initialize_wip_information(&mut self) -> (Epoch, HashSet<String>) {
+        // Hardcoded information about WIPs
+        let mut bits = vec![vec![]; 32];
+        let wip_0014 = BitVotesCounter {
+            votes: 0,
+            period: 26880,
+            wip: "WIP0014".to_string(),
+            init: 500000,
+            end: 850000,
+        };
+        bits[0].push(wip_0014);
+
+        // Assessment of new WIPs
+        let mut min_epoch = u32::MAX;
+        let mut old_wips = HashSet::default();
+        for (i, wips) in bits.into_iter().enumerate() {
+            for wip in wips {
+                if self.bit_tapi_counter.contains(i, &wip.wip) {
+                    old_wips.insert(wip.wip);
+                } else {
+                    if wip.init < min_epoch {
+                        min_epoch = wip.init;
+                    }
+                    self.bit_tapi_counter.insert(i, wip);
+                }
+            }
+        }
+
+        (min_epoch, old_wips)
+    }
+}
+
+/// Struct that count the positives votes of a WIP
+#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BitVotesCounter {
+    pub votes: u32,
+    pub period: Epoch,
+    pub wip: String,
+    pub init: Epoch,
+    pub end: Epoch,
+}
```

Comment:
I see fields here that imply unspecified TAPI features that are not covered by [WIP-0014](https://github.com/witnet/WIPs/blob/81346388b74e301e1a679365dae72d8042fbff7d/wip-0014.md).

Reply:
- `votes` is the counter
- `period` should be put to 26_880 (according to WIP-0014)
- `wip` is an identifier
- `init` and `end` are used to handle the duration of the vote process. Due that we are going to use the same bits for different vote processes, I think that if we do not specify, a node could activate automatically one WIP with votes from other WIP assigned in the same bit but in a later time

Reply:
I see. This kind of decisions, please raise them publicly. Otherwise we would end up implementing code that differs from the WIP or vice-versa.